src/level_four/core/controller/sync_cards_controller.c

```diff
@@ -103,7 +103,8 @@ void sync_cards_controller()
     case SYNC_CARDS_TAP_TWO_CARDS_FLOW:{
         tap_card_data.desktop_control = false;
         tap_threshold_cards_for_reconstruction_flow_controller(2);
-        if (counter.level == LEVEL_ONE || flow_level.level_two == LEVEL_THREE_WALLET_LOCKED) {
+        if (counter.level == LEVEL_ONE ||
+            (flow_level.level_one == LEVEL_TWO_OLD_WALLET && flow_level.level_two == LEVEL_THREE_WALLET_LOCKED)) {
```

Comment:
In tap_card_connection() function for [locked wallet](https://github.com/Cypherock/x1_wallet_firmware/blob/release/v0.4.768+/src/level_four/tap_cards/controller/controller_tap_cards.c#L225
), flow level one is not set to LEVEL_TWO_OLD_WALLET.

Change might also be required here in that function.